system/Config/Factories.php

```diff
@@ -110,12 +175,27 @@ public static function __callStatic(string $component, array $arguments)
             return null;
         }
 
-        self::$instances[$options['component']][$class]    = new $class(...$arguments);
-        self::$basenames[$options['component']][$basename] = $class;
+        self::$instances[$options['component']][$class] = new $class(...$arguments);
+        self::$basenames[$options['component']][$name]  = $class;
+
+        // If a short classname is specified, also register FQCN to share the instance.
+        if (! isset(self::$basenames[$options['component']][$class])) {
+            self::$basenames[$options['component']][$class] = $class;
+        }
 
         return self::$instances[$options['component']][$class];
     }
 
+    /**
+     * Is the component Config?
+     *
+     * @param string $component Lowercase, plural component name
+     */
+    private static function isConfig(string $component): bool
```

Comment:
This seems unnecessary as a method?

Reply:
This method is called from five locations within the class.
If we really don't need it, PhpStorm can refactor to make it inline.

Reply:
I'm fine with it, no harm.